### ISSUE TITLE
Gears of War 3 [4D5308AB]

### ISSUE THREAD

**Issue by FranciscoCarlosJunior** [xenia-canary/game-compatibility#39]
# Issues:
No issues noticed.

# Log:
[xenia.log](https://github.com/xenia-canary/game-compatibility/files/4519452/xenia.log)

# Screenshot(s):
![2020_04_22_19_27_33_1364x736](https://user-images.githubusercontent.com/60429667/80043998-34609e80-84d1-11ea-8ca0-21c23efe081c.jpg)
![2020_04_22_19_17_28_1364x736](https://user-images.githubusercontent.com/60429667/80044005-375b8f00-84d1-11ea-9b59-9f44c582140b.jpg)

# Labels:
state-gameplay

**Comment by smsmoosavi** [comment]
I got stuck in act 4 chapter 5 where 4 armored kantuss and a bunch of other enemies attack in 4 waves. I Killed all of them but nothing happened, the song keeps playing, teammates stay in cover, can't interact with the valve, ...
Restarting the chapter didn't work either. anybody got past this point?

**Comment by Typharius** [comment]
Looks perfect in terms of graphics, but stutters a lot.

**Comment by Lsn93** [comment]
It need more stability, emulator crashes often

**Comment by Silergo** [comment]
Broke script when defeat Berserk lambient. Cutscene doesn't start.

**Comment by mainzerbe** [comment]
Since build "1cf255f_canary_experimental" Gears 3 and Gears of Judgment stuck with "Loading profile" for me. A workaround is to delete the firewall settings in Win 10 but even then a second user is added although only one controller is connected.

**Comment by KnizP** [comment]
In first ingame animation it's crashing due to graphics device lost workaround for me was to change gpu = "vulcan" ,play problematic mission and switch back, same in chapter two.Sometimes problem with "objective lost" restaring chapter helps, but otherwise playble.

**Comment by Mal0-1471** [comment]
> In first ingame animation it's crashing due to graphics device lost workaround for me was to change gpu = "vulcan" ,play problematic mission and switch back, same in act two. Patches must be apply or game looks bad.Sometimes problem with "objective lost" restaring chapter helps, but otherwise playble.

Same Problem here, during the cutscene where you leave anvil gate, the game would crash and the same would occur if i skipped the cutscene, to fix this i had to play through the next level on Vulkan.

I also ran into the objective lost issue during a later level, restarting chapter does seem to fix this.

Other than that the game runs fantastic on my machine and looks great.

Edit: Came across a new problem in the last few levels.
Anytime the final boss is loaded into the level, if you die and reload checkpoint, the game will be bugged and you will occasionally teleport to the boss as well as the boss being unable to do any attacks, this of course breaks the final boss fight and makes it impossible. The only fix is to restart the entire chapter.

**Comment by heller390** [comment]
I am having audio issues like audio getting muted during cutscenes , NPCs very faint voice etc. Anyone have any solutions?

**Comment by Margen67** [comment]
@heller390 Tech support belongs on the Xenia [Discord server](https://discord.gg/Q9mxZf9)

**Comment by Mal0-1471** [comment]
Tested on July 29th build of Xenia Canary.

Played from start to finish in multiple sittings in split-screen co-op.

Entire game main campaign is playable, however, major problems occur if you load a checkpoint under specific circumstances.

The game seems to 'forget' what pawn it's going to run code on when an event happens, and auto-picks the player 1 character, this means that certain scripted objectives can cause a softlock, for instance, if you reload a checkpoint during the act 3 berserker fight, the game will instead only play the ending cutscene for that fight once player 1 dies, instead of the berseker. Later on, in act 5, theres a mortar enemy that when you get close gets given a loadout with just a boltok, but if you've reloaded this checkpoint, instead player 1's character is given the boltok and you cannot progress.

This bug makes it require you to reload the entire chapter in order to continue, making harder difficulties a unique challenge due to having to basically ignore checkpoints entirely.

Overall though, this emulates fine as long as you dont reload checkpoints due to a death or crash or whatever. Would be nice if it was fixed however.

**Comment by Sowa95** [comment]
Tested on: https://github.com/xenia-canary/xenia-canary/commit/a688f87f557ab34ae8f7ef1353fc608e9314490b

Playable from start to finish including RAAM's Shadow. Issue with checkpoints still present, restarting chapter helps.

**Comment by froggz19** [comment]
Gears 3 TU6 with all patches applied except bloom

RTV/DSV **extreme fps drop in certain areas**

![Image](https://github.com/user-attachments/assets/cb81b073-b7fe-4088-ae6f-810e5e797e5e)

ROV **stutter mess on first run but all smooth after that, stable 60 fps**

![Image](https://github.com/user-attachments/assets/a6602847-45c0-4b69-856a-274f388fbb09)

Windows 11 24H2 Build 26100.2894

**Comment by malh65521-png** [comment]
help how do i fix this 

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/dd38ab26-f9ef-4722-863d-ad37c96dfc41" />

**Comment by partomedico** [comment]
Enemies stop coming out and the objective Kantus Chapter 4 cannot be completed.
<img width="1872" height="1053" alt="Image" src="https://github.com/user-attachments/assets/378a89b1-92c5-4718-b6ac-75a7538c4632" />

**Comment by partomedico** [comment]
The error persists in the Gears of War 3 and Gears of War 3TU versions.
i7 8700k  - gtx 1080 - 32 ram

**Comment by RareMv** [comment]
> Enemies stop coming out and the objective Kantus Chapter 4 cannot be completed. <img alt="Image" width="1872" height="1053" src="https://private-user-images.githubusercontent.com/175161960/477932032-378a89b1-92c5-4718-b6ac-75a7538c4632.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyODg2NzMsIm5iZiI6MTc1NjI4ODM3MywicGF0aCI6Ii8xNzUxNjE5NjAvNDc3OTMyMDMyLTM3OGE4OWIxLTkyYzUtNDcxOC1iNmFjLTc1YTc1MzhjNDYzMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyN1QwOTUyNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NTlmZTZmY2Q0N2NlZDg3ZGI1MWZhOTRiMDQzZDdjOTQ3YTI3YjBkZWYyYTE0ZjhkYzEyYjhjMjYwZmQyZjFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.gpUIxwpDZgNeH16Gvvo0bAcSUjVXEn9kLR5DMA5zGWI">

Gears 3 will break scripting OFTEN times in Xenia if you ever die. You need to NEVER die to never break scripting basically. If you ever die, you need to restart the entire chapter for the scripting to work properly. Unfortunately Hard and Hardcore playthroughs in Xenia in single or multiplayer Gears 3 are not doable because of this. Or without great frustration.

**Comment by actuary87** [comment]
I have a good observation I hope it helps the developer to fix the annoying "objective lost" bug as described above.

I was able to play the game on Hard using Xenia (after repeating some chapters from the beginning, it was painful).

I then played on Insane using Xenia till I reached Act 1-4 "Helping Hand" until the "Get to Cougars Stadium" checkpoint where my character die randomly before I can clear the area and I cannot proceed further.
So I tried to copy the checkpoint to my Xbox to continue from there.

I noted that the same problem happens on the Xbox. I realized that the problem seems to be with the checkpoint data that Xenia writes (because the same bug continued on the Xbox).

On the other hand, when restarting the chapter on the Xbox and reaching to the same checkpoint, I was easily been able to clear it out and move on even AFTER DIYING FEW TIMES.

TL;DR
1. Xenia write checkpoint data differently compared to Xbox causing Xenia itself to not reload correctly on certain areas in the game.
2. Xbox write checkpoint data correctly and it does suffer from the savegame data written by Xenia (if tranferred to Xbox using Horizon) which let me question the issue is not Kismet Scripting Bug as we might think!

What I didn't try is what if I copy the clean checkpoint data (saved by Xbox) back to Xenia, will Xenia work correctly from that checkpoint even if I repeatedly died and tried to continue (because checkpoint is written once).

**Comment by partomedico** [comment]
I was able to fix it by setting the "Unlock FPS" to false in the patches, but later on the last boss I had an error but by restarting the chapter I was able to finish the campaign.

**Comment by actuary87** [comment]
I am not applying any patch for GoW3.

**Comment by mushrahmx** [comment]
I have an issue in various points of the campaign (even if you play on easy or insane) if you die in specific checkpoint game get bugged and you get an instakill, or get bugged out of tha map limits, i tried switching vulkan, or disabling patches but nothing works, the only solution is,  get the level flawless or reset the FULL CHAPTER and thats painful, is there a way i can fix this?

**Comment by RareMv** [comment]
No, needing to restart the entire chapter on death is currently a known limitation.

**Comment by evgeny-schalamov-lgtm** [comment]
The game runs great, but there are some issues in 4K. In some dreams, almost every location has issues even with 5090. The cutscenes are slow.
https://www.youtube.com/live/69AbqUYBRYU?si=zomuRsLQTM9pjdRH

**Comment by TheReaping** [comment]
WTF is wrong with this game? Why is it detecting two players with my controls mirrored? Why is the screen so distorted and terrible with only a small corner barely functioning? There are a lot of issues with this game, the "optimized" settings and patch did nothing.
<img width="3440" height="1440" alt="Image" src="https://github.com/user-attachments/assets/2b912f8f-4324-4d6f-9615-9a69b34e292b" /> 

<img width="3440" height="1440" alt="Image" src="https://github.com/user-attachments/assets/bca28017-872b-4599-b273-43ca3cb50418" />

EDIT: Figured out this garbage myself after a lot of testing.
DO NOT use Vulkan, use D3D12 instead. There ends up being green around some textures but at least it works.

DO NOT use SDL2 if you are using a PS4 controller. For some stupid reason it detects two controllers.

**Comment by soldiermxdeath** [comment]
I couldn't continue RAAM's Shadow due this error.
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/dead47b3-e10c-41fc-a8e0-8e7b8cc99cb8" />
I started the chapter with no death and this happened anyway. And the issue others reported if you die, the checkpoint is bugged and you need to restart the chapter, it happened to me too both main campaign and RAAM's Shadow. 

I also got this error randomly on both campaigns:
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/38158855-9961-4f3b-b0df-cf612843134c" />

Edit: I thought it was fixed with the recent xenia update, but in Raam shadow still persist the checkpoint issue after you use Raam on chapter 4, you die almost instantly when you start as Barrick, restarting checkpoint let you continue but is still an issue. I haven't replayed the main campaign yet.
Also the random GPU crash still persist.

**Comment by sk0tina** [comment]
Finished the game from start to finish on `xenia-canary@0af253e`.
No major graphical issues, one minor issue - the light sources are sometimes rendered through walls.
In terms of performance, there were some minor FPS drops (my guess that it's when the game loads the content), but other than that it was a stable 30 FPS.
The only game patch I used was "Show FPS".

All emulator settings in the config file were unchanged from default.

Played on:
CPU: Intel i5-10400f
GPU: Nvidia Geforce RTX 3060Ti
RAM: 32 GB DDR4
NVME SSD for both the game and the emulator

Regarding the scripting issues after reloading a checkpoint, they are still present. I combined everything that I personally encountered in a single list, most of them have already been reported in this issue.

1. Act 3 - Chapter 2, Rescue. At the end of the chapter you need to kill Lambent Berserker. After killing her, the cutscene should play but it doesn't, the Berserker just dies and nothing else happens.
2. Act 4 - Chapter 5, Bon Voyage(the one where you need to get the rotor and the fuel for the submarine). At the very end of the chapter you need to kill all enemies in the room, then use the valve to end the chapter. When you kill everyone, nothing happens, and when you try to use the valve, Markus says "It's not possible right now".
3. Act 5 - Chapter 4, Threshold. At the very start when you get outside, you need to fight in the big area with fountains, one of the enemies has mortar, and when you approach him, you are given Boltok pistol instead of him (as described previously by @Mal0-1471). After you clear the area, nothing happens.
4. Act 5 - Chapter 6, Reckoning. If you die while fighting the last boss, you will respawn floating in the air next to the arena, and with the boss seemingly inactive.
5. DLC - RAAM's Shadow - Chapter 2, Darkness Spreads. When you play as RAAM for the first time, if you die and restart a checkpoint, the second thumper will not start hammering as he should. Though unconfirmed since it's possible that I haven't waited long enough.
6. DLC - RAAM's Shadow - Chapter 5, Kryllstorm. When you fight the elite Theron in the canal, the fight will not finish properly if you restart and nothing happens when he dies. The cutscene should play after his death.

Restarting the chapter and finishing it without reloading a checkpoint (either manually from the menu or after dying) resolved every issue from the list.

**Comment by soldiermxdeath** [comment]
Random crashes specially on act 4 and 5 on many chapters like act 5 chapter 3, act 5 chapter 4, etc.

<img width="1919" height="1032" alt="Image" src="https://github.com/user-attachments/assets/43cc3d04-2177-46e4-b717-76d5215c946a" />
I'm using a RX 7900 XT, tested changing some gpu settings on config, changed dx12 to vulkan, nothing works.

**Comment by nfxw** [comment]
Also finished the game, but don't have the issues that was mentioned above by @sk0tina.
The only major issues are stuttering and crashing.
Finished in xenia edge version be10bbd, the game was updated to version 6.
1. At act 3 cutscene appears after killing Berserker, proof:  
<img width="1920" height="1200" alt="Image" src="https://github.com/user-attachments/assets/569c245c-367e-4d1b-a905-5f3cb44200c8" />
2. At act 4 valve will rotate normally, proof: 

https://github.com/user-attachments/assets/8d0ffbf4-2a9a-4e9e-bf58-eca7a811923b
3. At act 5 i don't understand  the issue, first the enemy using mortar, second pistol, everything is right.
https://github.com/user-attachments/assets/181e21c0-5460-40a1-9cdc-ace5612b27fb
4. Final boss is quite easy, so i don't get the same issue.
Also maybe all of this issue is because you don't have updated version of the game.

<img width="1920" height="1200" alt="Image" src="https://github.com/user-attachments/assets/f07fa8d6-a300-4bf8-a122-d00666f55f64" />

**Comment by nfxw** [comment]
> Don't change the subject, leave my mental ilness to me. I'm not saying you should be aware of every change, I'm saing you should test what you're supposed to be testing. If checkpoint issue was fixed on Edge, but not on Canary, you wouldn't even know you fucking donkey.
> 
> When you say "latest Canary" nobody knows what that means. It works today, it may not work tomorrow, or in 6 months from now. When visiting this motherfucking compatibility repo, it would be good to have some reference instead of guessing wtf did you use. Wild idea, I know. You don't bother to send log, or include title bar on the screenshot, so at least say which build (and you did this time, that was a compliment).
> 
> I knew you're a lost cause, but I didn't realize how bad of a lost cause until now.

I know that it's hard for you to understand such simple things, but i'll try again:
Latest canary means that you HAVE to look at the comment date and just compare to date of the latest build of canary.
For example: latest canary build was on october 21, my comment was on october 23, then it's MEANS that i'm using LATEST build which refferd to october 21.
As you can see i highlighted some words, so people like you can understand.
About log and version of the emulator i've talked before, so there is no reason to talk about it again.
Also i don't have time to arguing about logs, versions of the emulators, patches and other stuff, so don't waste my time about it.

**Comment by Sowa95** [comment]
Motherfucker it's your duty to provide the most basic information. You're dumb as a bag of rocks but don't assume everyone is equally dumb, I know you can check by date. Once you go through hoops to find the build you're using, that still just a build, we don't know anything else.

You remember GTA IV from few days ago? When you were spreading fake news that it's fixed, when it wasn't really fixed, and it wasn't even affected by Xenia changes? That's right, we had to guess everything because it wasn't repeatable, and your report alligned with Edge Vulkan changes that improve Nvidia issues (what a coencidence), even more guessing.

>Also i don't have time to arguing about logs, versions of the emulators, patches and other stuff, so don't waste my time about it.

We already know that. No testing, no config, no log, or empty log in dozens of your reports, no build, no title bar on the screenshot, no nothing. I guess I'm the dumb one for expecting any information about game's compatibility on a compatibility page. I should just accept "1440p no issues" on games riddled with issues.

**Comment by RareMv** [comment]
Gents, the necessary information has been requested, provided and understood by all parties. Please stay politely on topic.

**Comment by Sowa95** [comment]
>So the goal is to set the basic report, and then, if you or other people have more information, add it for more detailed report.

Don't worry. I'll fix it.

**Comment by nfxw** [comment]
> So you just want your name on the initial report for the bragging rights and credit, OK assumed so. Why else bring up how many games you moved to gameplay and playable? There are many "1440p no issues" or "plays fine" reports when they don't run well or need patches to get to gameplay. The problem with shit tier reports like this is other people read them, assume the game is working and when it doesn't someone else has to help them. You're just downloading a batch of games, getting to the first gameplay section and assuming it works. It's not helping anyone really.

I don't know where people like you or sowa came from.
I don't need a credit, i just move some issues from xenia-project(master) to canary, so it can at least be compared to rpcs3.
@Margen67 said that the game should be played from start to finish to go in playable and state-gameplay means functual gameplay, that's it, nothing more, other things like "reports like this is other people read them, assume the game is working and when it doesn't someone else has to help them" is just pathetic, people before mine and other repot go to compatibility list, see 200 games, don't see the game they wanted and go to rpcs3 instead, at least now they know that the game have functional gameplay.
I don't even want to start, that flickering issue with upscaling was fixed after 3 years of "development", async shader only present now in xenia edge, in rpcs3 it was emplenmented 8 years ago and other stuff and instead talk about this, you and that sowa bragging about compatibility reports instead of real problems.
And again, if you see problem with mine and other reports, get the game you wanted, play it as long as you want, so it's go to state-gameplay, then add some issues, log if needed, and that's it, you don't HAVE to write about it.
About other stuff like patches and anything else i don't even want to talk about, most of the games works fine without them.
I've heard that xbox emulation community is toxic but i don't even imagine it how big the problem is.

**Comment by Gliniak** [comment]
I completely removed this "conversation", because this isn't a place for such arguing.

My opinion (as a maintainer of this pile or garbage) is that:
- This is canary specific compability list, so every report (at least most visible part of it) should contain report from canary with specific hash of a version used. I understand that writing "latest" is easy way, but sometimes people report latest and then you look at log and it is build from 3 or 6 months ago or something like that. Providing hash is also beneficial when within one day you have multiple builds. 
- You CAN include report from edge or netplay, but that should be done ONLY after report from canary and under some specific label. Most reasonable is to make font bold and maybe increase size of it a bit and hid it under such dropdown.
- Reports should not be done with any custom patches applied and any TU that is used should be stated.
- If game have corrupted graphics that can be fixed with usage of some specific config option or by pressing F5 it should also be stated. I will create a label for that.

That's the end of a story. No need to shit on each other or how shitty this project is. I know it is shit, but that's something that can and will be at least partially addressed in the future. Remember that in comparation to RPCS3, Xenia DO NOT have basically anyone that would consider spending significant amount of time on it.

(I will update default template in the near future)
Example of how multiple reports should look like:

### Main canary report
 - Something, something...

<details>
<summary><h3>Edge report</h3></summary>
- Something, something...
</details>

<details>
<summary><h3>Netplay report</h3></summary>
- Something, something...
</details>

**Comment by dnmaia** [comment]
Im playing on Strix Halo(Radeon 8060S) and in my case, i see a huge improvement with minimal stutter when using D3D12.

**Comment by BorealisEcho** [comment]
As of fbd620c_canary_experimental (2026-05-30) release, this game benefits from ZPD occlusion queries.

Set `occlusion_query = "strict"` to fix unoccluded lens flares. "fast" is still better than "fake".

**Comment by lbacigal** [comment]
This has happened at least 5+ times while I've been playing, and it happens at the most random moments. The game freezes, I get that error message, and whatever audio was playing just loops infinitely. I have to close Xenia entirely to get the game working again. I had the same issue with both GoW 1 and 2 as well.

<img width="2559" height="1381" alt="Image" src="https://github.com/user-attachments/assets/856ff4bf-fc97-43c8-bd7f-c753662cdeef" />

Also I noticed sometimes the fire particles have orange squares around them
<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/ff4e5fbd-fab9-43bd-9f70-e86369c12c23" />

**Comment by Vovaline** [comment]
in-game cut-scenes are fine, but pre-made cut-scenes, such as "Previously on Gears", and other such cut-scenes of that style, are stuttering to single-digit frames.

**Comment by Sowa95** [comment]
https://github.com/user-attachments/assets/79d57af3-6feb-4c55-9452-d0779a239476

**Comment by n3xu589** [comment]
A shader bug occurs when scaling the image. I have attached three screenshots below showing different scaling levels:

2x2
<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/3311e948-11fc-477b-bbd2-be5e2b2d76a4" />

3x3
<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/1a969a86-b9cc-4759-8c79-9eafd1c8373e" />

4x4
<img width="5120" height="2880" alt="Image" src="https://github.com/user-attachments/assets/ac8cb90e-af1a-4332-a1aa-1cdb5611818f" />

As we can see, the higher the scaling factor, the better the final result. At 4x4, the shader bug disappears completely, and the image looks exactly as it should. This means that the game/emulator can, after all, correctly apply and display this shader.


As I understand it, this same shader is also used for ambient occlusion, since disabling ambient occlusion in the patches makes the bug disappear at any scale. But it also makes the image very flat.

And there is a choice: either to play with 4x4 scaling and a very low frame rate, or with 1x1 scaling and a very blurry image with very poor anti-aliasing.

Is there any possibility at all of fixing this bug in Xenia Canary in the foreseeable future?

**Comment by goldislead** [comment]
> ### 
> Is there any possibility at all of fixing this bug in Xenia Canary in the foreseeable future?

As of bd620c_canary_experimental, Gears 3 can be safely scaled without breaking post-processing or disabling effects via patch, provided a scaling threshold is set for any surfaces at or above 360 pixels wide. 

Set `draw_resolution_scale_threshold` to `364` (minimum) or `512` (safer).

**Comment by JuliusJoker92** [comment]
### Selective-scaling follow-up: possible detail loss across MSAA aliases

Following the [scaling-threshold advice above](https://github.com/xenia-canary/game-compatibility/issues/39#issuecomment-5064853506), we found a specific scale-classification problem and a small correction that may be useful here.

**Validation scope:** gameplay testing was performed in a local static-recompilation project using a Xenia-derived ReXGlue renderer, with Canary's selective-scaling implementation adapted from commit `74db632ab32e179dd0d6858b51caae07fc444949`. This is a downstream finding, **not a claim of a reproduction or verified fix in a current Xenia executable**. The same width-based classifier is present in the public Canary source inspected today, at [`IsScaleNativeForPitch`](https://github.com/xenia-canary/xenia-canary/blob/87c24112706d95f15f83dfec58e93923bd7ffa07/src/xenia/gpu/render_target_cache.cc#L589).

#### Observed result

For title `4D5308AB`, TU6, D3D12 host RTV targets, 2x2 scaling and `draw_resolution_scale_threshold=364`:

- Selective scaling removed green corruption with ambient occlusion, depth of field and post-processing enabled.
- However, the scene became visibly blocky despite 2560x1440 output and scaled main render targets.
- Instrumented intermediate-image captures showed detailed scene data followed by duplicated pixel blocks **before** a later post-processing draw. The loss therefore occurred earlier than that shader.

Transfer logs showed scene data passing through scaled depth, then a native-resolution 4x MSAA depth/stencil view, and subsequently returning to the scaled scene. The preserved EDRAM ranges included color data outside the nominal stencil effect area.

#### Why the classifier matters

The existing classifier calls `RenderTargetKey::GetWidth(pitch_tiles_at_32bpp, msaa_samples)`. With threshold 364, the same eight-tile row gets different scale classes depending on its MSAA interpretation:

| View | 32bpp tiles per row | Logical width | Existing classification |
| --- | ---: | ---: | --- |
| Single-sampled | 8 | 640 | Scaled |
| 4x MSAA | 8 | 320 | Native |

Our evidence points to the scaled-to-native transfer discarding high-resolution subpixels while preserving aliased data, then the return transfer duplicating the remaining data. A high-resolution allocation later in the frame cannot recover those lost subpixels.

#### Proposed correction for review

We changed the classification to use the tile-aligned **32bpp sample-row width**, independent of MSAA, preserving the existing disabled-scaling and host-target guards:

```cpp
bool RenderTargetCache::IsScaleNativeForPitch(
    uint32_t pitch_tiles_at_32bpp, xenos::MsaaSamples /*msaa_samples*/) const {
  uint32_t threshold = cvars::draw_resolution_scale_threshold;
  if (!threshold || !IsDrawResolutionScaled() ||
      GetPath() != Path::kHostRenderTargets) {
    return false;
  }
  uint32_t pitch_samples_tile_aligned =
      pitch_tiles_at_32bpp * xenos::kEdramTileWidthSamples;
  return pitch_samples_tile_aligned != 0 &&
         pitch_samples_tile_aligned <= threshold;
}
```

This makes both eight-tile views scaled at threshold 364, while genuinely small one-tile targets remain native. Configuration descriptions and comments should also reflect the changed units.

#### Downstream verification and limitations

- A compiled classifier harness using the before/after production function bodies, with minimal adapters, checked 206 row pitches across three MSAA modes and six thresholds, plus captured-view and disabled/ROV guards. The old classifier produced 31 violations of the tested MSAA-invariance/scene-preservation conditions; the corrected one produced zero.
- The downstream SDK built successfully.
- Subsequent corridor and combat captures retained sharp detail without green speckling. The person playing explicitly confirmed: “Sharp again, no green speckles.” Effects remained enabled.
- This does not establish that every cross-pitch or cross-format alias must share a scale class, nor that this heuristic is correct for every title or threshold. ROV and Vulkan gameplay were not validated by this change.
- The investigation was AI-assisted, using public renderer source and local instrumented captures. No game binaries, assets, generated game code or full GPU dumps are included here.

Could the maintainers check whether the equivalent alias transfer occurs in Canary, and whether preserving the scale class across these MSAA interpretations is appropriate? We wanted to share the concrete failure mechanism and candidate correction, while keeping the downstream validation boundary explicit.

**Comment by goldislead** [comment]
@JuliusJoker92 If you'd like to report a bug that's been _explicitly observed_ in Xenia Canary, please feel free to do so without the paragraphs of LLM text.
